synapse/lib/parser.py

```diff
@@ -514,16 +524,24 @@ def _larkToSynExc(self, e):
         elif isinstance(e, lark.exceptions.UnexpectedCharacters):  # pragma: no cover
             expected = sorted(set(terminalEnglishMap[t] for t in e.allowed))
             mesg += f'.  Expecting one of: {", ".join(expected)}'
-            at = e.pos_in_stream
-            line = e.line
-            column = e.column
+            soff = eoff = e.pos_in_stream
+            sline = eline = e.line
+            scol = ecol = e.column
         elif isinstance(e, lark.exceptions.UnexpectedEOF):  # pragma: no cover
             expected = sorted(set(terminalEnglishMap[t] for t in e.expected))
             mesg += ' ' + ', '.join(expected)
-            line = e.line
-            column = e.column
-
-        return s_exc.BadSyntax(at=at, text=self.text, mesg=mesg, line=line, column=column, token=token)
+            sline = eline = e.line
+            scol = ecol = e.column
+
+        texthash = hashlib.md5(self.text.encode(errors='surrogatepass'), usedforsecurity=False).hexdigest()
```

Comment:
We have **two** different ways that we are reporting the texthash; some methods use `errors='surrogatepass'` and others do not. We should reconcile those into a single function somewhere and make sure that our implementation lines up with what Optic may expect.

Reply:
Added a queryhash func to common.